From 617e6bea45f4606a7bb1af94b7c458acd4c7f6d6 Mon Sep 17 00:00:00 2001
From: "Tao, Jing" <jing.tao@intel.com>
Date: Wed, 11 Mar 2015 14:18:37 +0800
Subject: [PATCH 324/338] ov5693-ecs: return the max resolution

Return the maximum resolution when no suitable resolution can
be found.

Change-Id: I824a64a208bec640636ec8ce1effd508919ed2ed
Tracked-On: #H1204625895
Signed-off-by: Tao, Jing <jing.tao@intel.com>
Signed-off-by: Tao Wei <wei.tao@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/45791
---
 drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c b/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c
index 91c0587..8a1a49a 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c
@@ -1576,8 +1576,8 @@ static int ov5693_try_mbus_fmt(struct v4l2_subdev *sd,
 					fmt->height);
 	if (idx == -1) {
 		/* return the largest resolution */
-		fmt->width = ov5693_res[0].width;
-		fmt->height = ov5693_res[0].height;
+		fmt->width = ov5693_res[N_RES - 1].width;
+		fmt->height = ov5693_res[N_RES - 1].height;
 	} else {
 		fmt->width = ov5693_res[idx].width;
 		fmt->height = ov5693_res[idx].height;
-- 
1.9.1

